components/skeleton/nz-skeleton.component.ts

```diff
@@ -0,0 +1,125 @@
+import { Component, Input, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+import { NzSkeletonAvatar, NzSkeletonParagraph, NzSkeletonTitle } from './nz-skeleton.type';
+
+@Component({
+  selector: 'nz-skeleton',
+  templateUrl: './nz-skeleton.component.html',
+  host: {
+    '[class.ant-skeleton]': 'true',
+    '[class.ant-skeleton-with-avatar]': '!!nzAvatar',
+    '[class.ant-skeleton-active]': 'nzActive'
+  }
+})
+export class NzSkeletonComponent implements OnInit, OnChanges {
+  title: NzSkeletonTitle;
+  avatar: NzSkeletonAvatar;
+  paragraph: NzSkeletonParagraph;
+  avatarClassMap;
+  rowsList: number[] = [];
+  widthList: Array<number | string> = [];
+
+  @Input() nzActive = false;
+  @Input() nzLoading = true;
+  @Input() nzTitle: NzSkeletonTitle | boolean = true;
+  @Input() nzAvatar: NzSkeletonAvatar | boolean = false;
+  @Input() nzParagraph: NzSkeletonParagraph | boolean = true;
+
+  getTitleBasicProps(hasAvatar: boolean, hasParagraph: boolean): NzSkeletonTitle {
```

Comment:
You don't need to expose these `get*` methods as `public`. 

You'd better write them into `updateProps` since they're only used there, to improve code clarity. And in that way you don't have to pass these parameters all over the place.